### PR TITLE
feat: transform entrypoints and add library type

### PR DESCRIPTION
**Pull request by muellerdemos**
This PR enables the transformation of the entry points object which is defined in the webpack config when creating an addon.
This way we keep the simple usage but we will create a browser digestible output of addon files.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,34 +18,47 @@ function resolve (dir) {
...
   return {
     entry: entrypoints,
     mode: isProduction ? 'production' : 'development',
     output: {
       path: resolve('dist'),
       filename: `[name].umd.js`,
-      library: `${addon_name}`,
-      libraryTarget: 'umd',
-      libraryExport: 'default'
+      library: `${addon_name}`
```
Comment (spiess-demos) [inline src/index.js:61]: Nitpicky question: wouldn't it be enough to assign the string like so?

```suggestion
      library: addon_name
```

File: src/index.js [hunk 2]
```diff
@@ -18,34 +18,47 @@ function resolve (dir) {
...
+   * so a browser is able to digest the output but still keep
+   * the simple webpack config mentioned above to create addons
+   */
+  for (const key of Object.keys(entrypoints)) {
+    entrypoints[key] = {
+      import: entrypoints[key]
+    }
+    entrypoints[key]['library'] = {
+      name: key,
+      type: 'window'
```
Comment (spiess-demos) [inline src/index.js:51]: Maybe first-time readers of these lines will get a jump-start when mentioning that you're utilizing the EntryDescription object alongside configuring library options per entry point, as seen in the [Webpack documentation](https://webpack.js.org/configuration/output/#outputlibrary), within a short comment.

Comment (muellerdemos) [reply src/index.js:55]: Added a more verbose comment here: https://github.com/demos-europe/demosplan-js-addon/pull/14/commits/4163c2a6946ddb6b09f8928c48dc0e210af22347

**Review by graupnerdemos — APPROVED** [review]
Thanks a lot!

**Pull request merged**